chainntnfs/txnotifier.go

```diff
@@ -192,18 +192,22 @@ func (r ConfRequest) ConfHintKey() ([]byte, error) {
 // the outputs of the transaction to determine if it matches. Otherwise, we'll
 // match on the txid.
 func (r ConfRequest) MatchesTx(tx *wire.MsgTx) bool {
-	if r.TxID != ZeroHash {
-		return r.TxID == tx.TxHash()
+	scriptMatches := func() bool {
+		pkScript := r.PkScript.Script()
+		for _, txOut := range tx.TxOut {
+			if bytes.Equal(txOut.PkScript, pkScript) {
+				return true
+			}
+		}
+
+		return false
 	}
 
-	pkScript := r.PkScript.Script()
-	for _, txOut := range tx.TxOut {
-		if bytes.Equal(txOut.PkScript, pkScript) {
-			return true
-		}
+	if r.TxID != ZeroHash {
+		return r.TxID == tx.TxHash() && scriptMatches()
```

Comment:
+1 for a stricter check

Will also need to do a sweep to ensure that we're properly setting the script everywhere. I _think_ we did this properly once we switched to the current script based dispatch, but won't hurt to double check.

Reply:
double checked all the call sites and appears we are